# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/cubeb-coreaudio/run_sanitizers.sh
# Commit: 24b97b0c24f0
# Full Hash: 24b97b0c24f0f30965d5d30dfa2a796e3c994e5e
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2020-02-13 21:42:57
# Description:
#   Bug 1613156 - Update cubeb-coreaudio to 7fe03b4. r=padenot
#   
#   Pick commits:
#   - 7fe03b4: Bail out if the output device has no output channel (#50)
#   - ad74bad: Update README
# ==============================================================================

diff -r 8e18b9d4caae -r 24b97b0c24f0 third_party/rust/cubeb-coreaudio/run_sanitizers.sh
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/cubeb-coreaudio/run_sanitizers.sh	Thu Feb 13 15:52:30 2020 +0000
@@ -0,0 +1,27 @@
+# The option `Z` is only accepted on the nightly compiler
+# so changing to nightly toolchain by `rustup default nightly` is required.
+
+# See: https://github.com/rust-lang/rust/issues/39699 for more sanitizer support.
+
+toolchain=$(rustup default)
+echo "\nUse Rust toolchain: $toolchain"
+
+if [[ $toolchain == nightly-* ]]
+then
+   echo "Run sanitizers!"
+else
+   echo "The sanitizer is only available on Rust Nightly only. Skip."
+   exit
+fi
+
+echo "\n\nRun ASan\n-----------\n"
+RUSTFLAGS="-Z sanitizer=address" sh run_tests.sh
+
+echo "\n\nRun LSan\n-----------\n"
+RUSTFLAGS="-Z sanitizer=leak" sh run_tests.sh
+
+echo "\n\nRun MSan\n-----------\n"
+RUSTFLAGS="-Z sanitizer=memory" sh run_tests.sh
+
+echo "\n\nRun TSan\n-----------\n"
+RUSTFLAGS="-Z sanitizer=thread" sh run_tests.sh